hudi-utilities/src/test/java/org/apache/hudi/utilities/sources/helpers/TestKafkaSourceUtil.java

```diff
@@ -63,4 +66,23 @@ void testConfigureSchemaDeserializer() {
     String schemaHash = Base64.encode(HashID.hash(avroSchemaJson, HashID.Size.BITS_128));
     assertEquals(props.getString(NATIVE_KAFKA_CONSUMER_GROUP_ID, ""), schemaHash);
   }
+
+  @Test
+  public void testFilterKafkaParameters() {
+    Map<String, Object> kafkaParams = new HashMap<>();
+
+    kafkaParams.put("custom1.config.streamer", "offer");
+    kafkaParams.put("boostrap.servers", "dns:port");
+    kafkaParams.put("custom2.config.capture", "s3://folder1");
+    kafkaParams.put("custom1config.sourceprofile.refresh.mode", "ENABLED");
+    // Case 1: No prefixes are configured.
+    assertEquals(kafkaParams, KafkaSourceUtil.filterKafkaParameters(kafkaParams, ""));
+    // Case 2: Ensure only the appropriate configs are filtered out.
+    Map<String, Object> filteredParams = KafkaSourceUtil.filterKafkaParameters(kafkaParams, "custom1,custom2");
+    Map<String, Object> expectedParams = new HashMap<>();
+    expectedParams.put("boostrap.servers", "dns:port");
+    assertEquals(expectedParams, filteredParams);
```

Comment:
just 1 additional ask. 
can you also test "config2." to ensure its not regex matching, but exact prefix matching.